http://bugs.gentoo.org/294470

this is two commits smooshed into one

From f3ed03373affbd13ee8dbbeb933959381ae492bb Mon Sep 17 00:00:00 2001
From: Matt Turner <mattst88@gmail.com>
Date: Wed, 17 Feb 2010 09:52:26 -0500
Subject: [PATCH] Fix SOCK_CLOEXEC on HPPA to match O_CLOEXEC.

The kernel defines SOCK_CLOEXEC as
include/linux/net.h:#define SOCK_CLOEXEC        O_CLOEXEC
arch/parisc/include/asm/fcntl.h:#define O_CLOEXEC   010000000

But glibc was defining
sysdeps/unix/sysv/linux/hppa/bits/socket.h:  SOCK_CLOEXEC = 01000000,

And even older glibc was doing:
sysdeps/unix/sysv/linux/hppa/bits/socket.h:  SOCK_CLOEXEC = 02000000,

Signed-off-by: Matt Turner <mattst88@gmail.com>
Signed-off-by: Kyle McMartin <kyle@redhat.com>
Signed-off-by: Carlos O'Donell <carlos@codesourcery.com>
---
 ChangeLog.hppa                             |    5 +++++
 sysdeps/unix/sysv/linux/hppa/bits/socket.h |    2 +-
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/hppa/bits/socket.h b/sysdeps/unix/sysv/linux/hppa/bits/socket.h
index bcc27cc..819b398 100644
--- ports/sysdeps/unix/sysv/linux/hppa/bits/socket.h
+++ ports/sysdeps/unix/sysv/linux/hppa/bits/socket.h
@@ -62,7 +62,7 @@ enum __socket_type
   /* Flags to be ORed into the type parameter of socket and socketpair and
      used for the flags parameter of paccept.  */
 
-  SOCK_CLOEXEC = 02000000,	/* Atomically set close-on-exec flag for the
+  SOCK_CLOEXEC = 010000000,	/* Atomically set close-on-exec flag for the
 				   new descriptor(s).  */
 #define SOCK_CLOEXEC SOCK_CLOEXEC
 #undef SOCK_NONBLOCK
-- 
1.7.1

